File: getting-started/src/main/java/it/marchino/quarkus/GreetingResource.java
package it.marchino.quarkus;

import javax.inject.Inject;
import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.PathParam;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

import it.marchino.quarkus.service.GreetingService;

@Path("/hello")
public class GreetingResource {
	@Inject GreetingService greetingService;

	@GET
	@Produces(MediaType.TEXT_PLAIN)
	public String hello() {
		return greetingService.sayHello();
	}

	@GET
	@Produces(MediaType.TEXT_PLAIN)
	@Path("/{name}")
	public String hello(@PathParam("name") String name) {
		return greetingService.sayHello(name);
	}

}

File: getting-started-kotlin/src/main/kotlin/it/marchino/quarkus/GreetingResource.kt
package it.marchino.quarkus

import javax.ws.rs.GET
import javax.ws.rs.Path
import javax.ws.rs.Produces
import javax.ws.rs.core.MediaType
import javax.inject.Inject
import javax.ws.rs.PathParam

@Path("/hello")
class GreetingResource {

  @Inject
  lateinit var greetingService: GreetingService

  @GET
  @Produces(MediaType.TEXT_PLAIN)
  fun hello() = "Hello RESTEasy"

  @GET
  @Path("/greeting/{name}")
  @Produces(MediaType.APPLICATION_JSON)
  fun greet(@PathParam("name") name: String): Greeting = Greeting(greetingService.greet(name));
}
